File: 203-remove-linked-list-elements/remove-linked-list-elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {

      if(head ==NULL) return NULL;

        ListNode* dummy = new ListNode(-1);
        dummy->next= head;
        ListNode* curr= dummy;

        while(curr->next != NULL){
            if(curr->next->val == val){
                ListNode* next_next= curr->next->next;
                 delete(curr->next);
                curr->next= next_next;
               
            }else{
                curr=curr->next;
            }
        }
return dummy->next;


    }
};

File: 203-remove-linked-list-elements/remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        // Create a dummy node that points to head
        ListNode dummy = new ListNode(-1);
        dummy.next = head;
        
        // Start with dummy node
        ListNode curr = dummy;
        
        // Traverse the list
        while (curr.next != null) {
            if (curr.next.val == val) {
                // Remove the node by skipping it
                curr.next = curr.next.next;
            } else {
                // Move to the next node
                curr = curr.next;
            }
        }
        
        // Return the head of the modified list
        return dummy.next;
    }
}